lib/lf/CMakeLists.txt

```diff
@@ -6,4 +6,5 @@ add_subdirectory(mesh)
 add_subdirectory(quad)
 add_subdirectory(refinement)
 add_subdirectory(uscalfe)
+add_subdirectory(intergridfe)
```

Comment:
Intergrid transfers may later also be needed for other finite element spaces beyond Lagrangian FE. See also my suggestions to move the `MeshFunction` capabilities out of the `uscalfe` namespace.

Reply:
Sorry for my late response, I was on Holidays for two weeks and then Christmas started almost immediately...
I would support to move most of the mesh function related functionality into it's own namespace/folder.
However, I'm not a big fan of moving the mesh function capabilities into `intergridfe` because that would suggest that this namespace is only concerned with "intergrid" functionality and this is very close to the functionality in the `lf::refinement` namespace. 

I suggest that the new home should contain all mesh function related classes that depend only on `lf::mesh`, `lf::mesh::utils`, `lf::base`. E.g. the class `MeshFunctionFE` would remain in `lf::uscalfe` because it depends on other classes from `lf::uscalfe` (e.g. `UniformScalarFESpace`). Also we could move the class `MeshFunctionTransfer` into `lf::refinement` because it depends on `MeshHierarchy`. But universal mesh functions such as `MeshFunctionGlobal` or `MeshFunctionConstant` or all the operator overloads would be moved to the new location.

Now I have two ideas for the new home for the mesh functions:
1) `lf::mesh::utils` because it already contains `MeshDataSet` and it's derived classes.  The MeshFunction concept is very similar to a MeshDataSet so it would make sense to bundle them in the same namespace. Also one can argue that a MeshFunction is really a utility "concept" based on the mesh interface.
2) `lf::mf` because quite a number of classes and not every user of `lf::mesh::utils` also wants automatically include all those classes. Also the MeshFunction classes would not be mixed with all the MeshBuilder classes from `lf::mesh::utils`.

I would prefer solution 1. @hiptmair What do you think? Do you have another idea?

Reply:
Both options have pros and cons. I have no objections to **1.**.

Reply:
I've moved all the mesh function now over to `lf::mesh::utils`, see PR #170. I would suggest to merge PR #170  before this PR into master.  @hiptmair can you have a quick look at #170 (most of it is just boring namespace adjustments ;)